packages/router/src/__tests__/setContextReuse.test.tsx

```diff
@@ -87,7 +87,7 @@ test("Doesn't destroy <Set> when navigating inside, but does when navigating bet
   act(() => navigate(routes.ctx3()))
   await waitFor(() => screen.getByText('3-initialSetValue'))
   act(() => navigate(routes.ctx4()))
-  await waitFor(() => screen.getByText('4-initialSetValue'))
+  await waitFor(() => screen.getByText('4-initialSetValue')) // Because its wrapping the same context provider, it should retain the value
```

Comment:
@KrisCoulson I had to change this test here. Would you double check that this also makes sense to you?

We just implemented "sharing" wrapped components, so shouldn't the context be shared as well? 

In this case `routes.ctx3` has a single wrap with the context, and so does `routes.ctx4`. Wouldn't it be desired behaviour to retain the value updated in ctx3?

Reply:
Sorry, referring to this line here: https://github.com/redwoodjs/redwood/pull/9299/files#diff-fce6235b482da06b280b4a95af99894d944b96548a5362c01f0ac2fa3973dc64R90